packages/quantic/force-app/main/default/lwc/quanticResultAction/quanticResultAction.js

```diff
@@ -0,0 +1,149 @@
+import {LightningElement, api} from 'lwc';
+
+/** @typedef {import("coveo").Result} Result */
+
+/**
+ * The `QuanticResultAction` component allows the end user to perform a specific action on a result.
+ * @category Search
+ * @category Insight Panel
+ * @example
+ * <c-quantic-result-action result={result} event-name={attachtocase} icon-name="utility:chat" label="Attach result" selected label-when-on="Detach result"></c-quantic-result-action>
+ */
+export default class QuanticResultAction extends LightningElement {
+  /**
+   * The label displayed in the tooltip of the result action button.
+   * @api
+   * @type {string}
+   */
+  @api label;
+  /**
+   * The label displayed in the tooltip when the result action button is selected.
+   * @api
+   * @type {string}
+   */
+  @api labelWhenOn;
+  /**
+   * The name of the icon displayed inside the result action button.
+   * @api
+   * @type {string}
+   */
+  @api iconName;
+  /**
+   * Specifies whether the result action button is in selected state or not.
+   * @api
+   * @type {boolean}
+   */
+  @api selected;
+  /**
+   * Specifies whether the result action button is in loading state or not.
+   * @api
+   * @type {boolean}
+   */
+  @api loading;
+  /**
+   * The name of the event to be dispatched.
+   * @api
+   * @type {string}
+   */
+  @api eventName;
+  /**
+   * The result to perform the action on.
+   * @api
+   * @type {Result}
+   */
+  @api result;
+
+  /** @type {string} */
+  resultActionOrderClasses;
+  /** @type {boolean} */
+  _selected;
+  /** @type {boolean} */
+  _loading;
+
+  connectedCallback() {
+    this.setSelected(this.selected);
+    this.setLoading(this.loading);
+
+    const resultActionRegister = new CustomEvent(
+      'qunatic__resultactionregister',
+      {
+        bubbles: true,
+        composed: true,
+        detail: {
+          applyCssOrderClass: this.applyCssOrderClass,
+        },
+      }
+    );
+    this.dispatchEvent(resultActionRegister);
+  }
+
+  /**
+   * Applies the proper CSS order class.
+   * @param {'first' | 'middle' | 'last'} order
+   */
+  applyCssOrderClass = (order) => {
+    const commonButtonClass = 'result-action_button';
+    let orderClass = '';
+
+    if (order === 'first') {
+      orderClass = 'result-action_first';
```

Comment:
is it possible to do this with css directly using selectors like `:last-child` and `:first-child`? maybe that would be easier.

Reply:
The reason I couldn't use CSS directly to address this is the following:
https://github.com/coveo/ui-kit/pull/2550#discussion_r1023173920

Reply:
Ah okok my bad thanks